src/appsignal/cli/diagnose.py

```diff
@@ -0,0 +1,390 @@
+# ruff: noqa: E501
+
+from __future__ import annotations
+
+import json
+import os
+import platform
+import urllib
+from argparse import ArgumentParser
+from typing import Any
+
+import requests
+
+from appsignal.agent import Agent
+from appsignal.config import Config
+from appsignal.push_api_key_validator import PushApiKeyValidator
+
+from ..__about__ import __version__
+from .command import AppsignalCLICommand
+
+
+class AgentReport:
+    def __init__(self, report: dict) -> None:
+        self.report = report
+
+    def extension_loaded(self) -> str:
+        return self.report["boot"]["started"]["result"]
+
+    def configuration_valid(self) -> str:
+        if self.report["config"]["valid"]["result"]:
+            return "valid"
+        return "invalid"
+
+    def configuration_error(self) -> str | None:
+        if self.configuration_valid() == "invalid":
+            return self.report["config"]["valid"]["error"]
+        return None
+
+    def started(self) -> str:
+        if self.report["boot"]["started"]["result"]:
+            return "started"
+        return "not started"
+
+    def user_id(self) -> str:
+        return self.report["host"]["uid"]["result"]
+
+    def group_id(self) -> str:
+        return self.report["host"]["gid"]["result"]
+
+    def logger_started(self) -> str:
+        if "logger" in self.report and self.report["logger"]["started"]["result"]:
+            return "started"
+        return "not started"
+
+    def working_directory_user_id(self) -> bool:
+        return (
+            "working_directory_stat" in self.report
+            and self.report["working_directory_stat"]["uid"]["result"]
+        )
+
+    def working_directory_group_id(self) -> bool:
+        return (
+            "working_directory_stat" in self.report
+            and self.report["working_directory_stat"]["gid"]["result"]
+        )
+
+    def working_directory_permissions(self) -> bool:
+        return (
+            "working_directory_stat" in self.report
+            and self.report["working_directory_stat"]["mode"]["result"]
+        )
+
+    def lock_path(self) -> str:
+        if "lock_path" in self.report and self.report["lock_path"]["created"]["result"]:
+            return "writable"
+        return "not writable"
+
+
+class DiagnoseCommand(AppsignalCLICommand):
+    @staticmethod
+    def init_parser(parser: ArgumentParser) -> None:
+        parser.add_argument(
+            "--send-report",
+            action="store_true",
+            help="Send the report to AppSignal",
+        )
+        parser.add_argument(
+            "--no-send-report",
+            action="store_true",
+            help="Do not send the report to AppSignal",
+        )
+
+    def run(self) -> int:
+        agent = Agent()
+        agent_json = json.loads(agent.diagnose())
+        self.config = Config()
+        self.send_report = self.args.send_report
+        self.no_send_report = self.args.no_send_report
```

Comment:
I think it's worth throwing an error and bailing out early if both `--send-report` and `--no-send-report` are specified. Otherwise, we'll do weird things later (e.g. printing both "Sending this report" and "Not sending this report" messages)

Additionally, note that `argparse` has this `BooleanOptionalAction` that implements `--option` and `--no-option` for you automatically. Unfortunately, if I'm reading the docs right, it's only implemented in Python 3.9 and above. :(

(You'll have to search for it manually [in the `add_argument()` docs](https://docs.python.org/3/library/argparse.html#action), I can't link to it directly because Python's documentation is entirely composed of unstructured streams of consciousness)